lib/db.js

```diff
@@ -84,7 +92,57 @@ class LimitDBRedis extends EventEmitter {
     });
   }
 
+  setupPing(pingConfig) {
+    if (!pingConfig || !pingConfig.enabled) {
+      return;
+    }
+
+    this.failedPings = 0;
+    this.once('ready', () => {
+      this.startPing(pingConfig);
+    })
+  }
+
+  startPing(config) {
+    this.keepPingRunning = true;
+    const doPing = () => {
+      this.redis.ping((err, duration, payload) => {
+        if(!err) {
+          this.reconnecting = false
+          this.failedPings = 0;
+          this.emit('ping - success');
+        } else {
+          this.failedPings++;
+          
+          if (this.failedPings >= config.maxFailedAttempts) {
+            if (config.reconnectIfFailed && !this.reconnecting) {
+              this.reconnecting = true
+              this.failedPings = 0;
+              this.redis.disconnect(true);
+              this.emit('ping - reconnect attempted', err);
+            } else {
+              this.emit('ping - fail', err);
```

Comment:
```suggestion
              this.emit('ping - error', err);
```
I'd suggest to use `ping - error`, more consistent with the other `node error`. 
Should we add the taken time?